### ISSUE TITLE
Fails on repeating identical deployment - stemcell already exists, not in gzip format

### ISSUE THREAD

**Issue by xtreme-gavin-enns** [concourse/bosh-deployment-resource#20]
![screen shot 2016-02-09 at 4 54 48 pm](https://cloud.githubusercontent.com/assets/1178694/12932832/edc56178-cf52-11e5-9850-962ec625ab92.png)

This screenshot is from our pipeline to redeploy concourse. There are no tasks in the pipeline, only resource gets and puts. We have a job to collect versions, so we can be certain that this deployment was using identical resources to the deployment that succeeded less than a minute before-hand. It continues to fail a day later with identical inputs.

**Comment by vito** [comment]
Are you using a github release resource with an access token configured, by any chance?

**Comment by xtreme-gavin-enns** [comment]
we are

**Comment by vito** [comment]
I'm betting it's because of this, then; just ran into a similar issue myself:

https://github.com/google/go-github/issues/246

**Comment by xtreme-gavin-enns** [comment]
interesting. hard to explain why it succeeded with the first deploy then failed with what should have been a no-op deploy about one minute later though. 
![screen shot 2016-02-10 at 3 12 35 pm](https://cloud.githubusercontent.com/assets/1178694/12960339/c7962b0c-d008-11e5-878b-7bc9e379932a.png)
![screen shot 2016-02-10 at 3 12 10 pm](https://cloud.githubusercontent.com/assets/1178694/12960340/c79941ca-d008-11e5-8fcd-4a3a44fbd413.png)

**Comment by vito** [comment]
Yep. To confirm you could hijack in and `cat` the release tarballs.

I'm starting to suspect this was a change rolled out by AWS or some other external service (GitHub?). Or maybe you didn't have an access token configured before?

**Comment by xtreme-gavin-enns** [comment]
I'm fairly certain the configuration was identical between those two builds - you can't see it in the screenshot but they occurred within a minute of each other.

And yes, your hunch is correct - the tgz files in the resource directory actually contain error xml

**Comment by xtreme-gavin-enns** [comment]
```Error><Code>InvalidArgument</Code><Message>Only one auth mechanism allowed; only the X-Amz-Algorithm query parameter, Signature query string parameter or the Authorization header should be specified</Message><ArgumentName>Authorization</ArgumentName><ArgumentValue>Bearer...```

**Comment by vito** [comment]
This should be fixed now.

**Comment by sunjayBhatia** [comment]
We ran into this same error from the https://github.com/concourse/github-release-resource/ when the repo that had the releases on it was moved to another organization. It seems related but we weren't able to track down the bug if it still exists in the go-github package. We just wanted to mention it somewhere in case someone else stumbles on the same problem. The fix for us was just updating the resource in our pipeline to the new organization name so it didn't hit the redirect anymore.
